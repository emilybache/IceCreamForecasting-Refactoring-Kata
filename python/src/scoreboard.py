import datetime

import scorer
from scorer import IceCream


def print_scoreboard():
    names = ["Steve", "Julie", "Francis"]
    now = datetime.datetime.now()
    print(f"Scoreboard at time {now}")
    for name in names:
        print_score(name)


def print_score(name):
    if name == "Steve":
        scorer.flavour = IceCream.Strawberry
    else:
        scorer.update_selection()
    score = scorer.get_score()
    print(f"{name} score: {score}")
