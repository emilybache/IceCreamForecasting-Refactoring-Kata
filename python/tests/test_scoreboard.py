from scoreboard import print_scoreboard


def test_scoreboard():
    assert print_scoreboard() == ""
