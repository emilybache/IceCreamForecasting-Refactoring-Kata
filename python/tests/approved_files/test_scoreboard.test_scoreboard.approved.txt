Scoreboard at time 2023-05-08 10:12:37.023834
Steve score: 5
Julie score: 6
Francis score: 5
